Code/Framework/AzToolsFramework/AzToolsFramework/API/PythonLoader.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include <AzToolsFramework/API/PythonLoader.h>
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Math/Sha1.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzCore/std/string/tokenize.h>
+#include <AzCore/Settings/ConfigParser.h>
+#include <AzCore/Utils/Utils.h>
+
+namespace AzToolsFramework::EmbeddedPython
+{
+    PythonLoader::PythonLoader()
+    {
+        LoadRequiredModules();
+    }
+
+    PythonLoader::~PythonLoader()
+    {
+        UnloadRequiredModules();
+    }
+
+    AZ::IO::FixedMaxPath PythonLoader::GetPythonVenvPath(AZ::IO::PathView thirdPartyRoot, AZStd::string_view engineRoot)
+    {
+        // Perform the same hash calculation as cmake/CalculateEnginePathId.cmake
+
+        // Prepare the engine path the same way as cmake/CalculateEnginePathId.cmake
+        AZStd::string enginePath = AZ::IO::FixedMaxPath(engineRoot).StringAsPosix();
+        enginePath += '/';
+        AZStd::to_lower(enginePath.begin(), enginePath.end());
```

Comment:
It was to match the "${CMAKE_CURRENT_SOURCE_DIR}/" argument in [LYPthon.cmake](execute_process(COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CalculateEnginePathId.cmake "${CMAKE_CURRENT_SOURCE_DIR}/"). But I can remove it from here and in LYPthon.cmake file as well.

Reply:
It turns out with you normalize a path is the parent ellipses: ` D:\github\o3de\python\..` with `cmake_path(NORMAL_PATH,...)` , it normalizes with the trailing `/`. If you normalize a normal path without the trailing `/', there won't be a trailing `/` added to it. We use the ellipses in [python.sh](https://github.com/spham-amzn/o3de/blob/9759f5d632c83855b947585efd822d5e6f88c332/python/python.sh#L46-L49) to determine the root engine folder. So it was simpler to make it consistent this way, since normalizing something with multiple '//' will still return '/'.